pkg/server/image_pull.go

```diff
@@ -331,13 +331,29 @@ func (c *criContainerdService) pullImage(ctx context.Context, rawRef string, aut
 	// Use config digest as imageID to conform to oci image spec, and also add image id as
 	// image reference.
 	imageID := configDesc.Digest.String()
-	if err := c.imageStoreService.Put(ctx, imageID, desc); err != nil {
-		return "", "", "", fmt.Errorf("failed to put image id %q into containerd image store: %v",
-			imageID, err)
+	if err := c.createImageReference(ctx, imageID, desc); err != nil {
+		return "", "", "", fmt.Errorf("failed to update image id %q: %v", imageID, err)
 	}
 	return imageID, repoTag, repoDigest, nil
 }
 
+// createImageReference creates image reference inside containerd image store.
+func (c *criContainerdService) createImageReference(ctx context.Context, name string, desc imagespec.Descriptor) error {
```

Comment:
Per discussion in slack consider TODOs to figure out which is the more performant sequence create then update or update then create; and to resolve a possible issue of removing wrong ":latest" target image.